### DISCUSSION TITLE
Windows vs Mac Caching

### DISCUSSION THREAD

**Discussion by joshualeond** [prefix-dev/pixi#1160] [Q&A] [answered]
Hi there! As a Julia user, I'm super excited to see this package manager coming to Python!

I have a question about caching. When I use `pixi` on my Windows device I see that it creates a `.pixi` folder within my project. This folder usually ends up containing thousands of files and is > 1 GB for smaller projects:

![image](https://github.com/prefix-dev/pixi/assets/3802502/da9a960a-a066-47e3-b669-78aa10fd590c)

To me it looks like it's not actually using the cached directory that I see in the `pixi info`: `C:\Users\myusername\AppData\Local\rattler/cache`.

When I use `pixi` on my Mac I'm not seeing this `.pixi` folder anywhere within the individual project.

Is this how it is designed to function or have I done something wrong on the user side?

Thanks!

**Comment by ruben-arts** [comment]
Hello,

After a `pixi install` there should always be a .pixi folder in the project root folder. On all platforms we support. 
The 1GB that you see should be linked to the cache.

Edit, reading on my phone isn't a good idea...

I'll check the cache dirs on my Windows machine tomorrow!

 On your Mac the . Folders might be hidden.

Hopefully this helps

**Reply by joshualeond** [reply]
Hi! This does help, I see now that it was just hidden on my Mac. Is there documentation around how this caching/linking works? I'm still a bit confused as to why the local `.pixi` folders grow so large. Thanks!

**Reply by ruben-arts** [reply]
We don't have a lot of documentation that really explains the logic in depth. Here is a short portion on the cache https://pixi.sh/latest/features/environment/#caching

**Reply by joshualeond** [reply]
Thank you! I dived a little deeper and I see the `conda-meta` directory with the `json` files that point to the extracted package in the cache. 

Does `pixi` then essentially copy/paste these extracted files from the cache into the local `.pixi` directory?

**Reply by ruben-arts** [reply, marked as answer]
Apart from some specific files that need to be modified to install. It should either hard-link or reflink the files into the .pixi folder. I'm still trying to find a proper tool to really visualize the actual data on your device taken up by pixi.

**Reply by joshualeond** [reply]
Oh okay, apologies for the confusion. I was able to test this by looking into one of my local `.pixi` directories and running the following command against something within the folder (like `python.exe`):

```cmd
fsutil hardlink list python.exe
```
It then showed me all of the hardlinks:

```
\Users\myusername\AppData\Local\rattler\cache\pkgs\python-3.12.2-h2628c8c_0_cpython\python.exe
\Users\myusername\Documents\stan-test\.pixi\envs\default\python.exe
```

I get this better now. Thank you for the response.
